packages/astro/src/core/middleware/loadMiddleware.ts

```diff
@@ -1,16 +1,22 @@
 import type { ModuleLoader } from '../module-loader/index.js';
 import { MIDDLEWARE_MODULE_ID } from './vite-plugin.js';
+import type { Logger } from '../logger/core.js';
 
 /**
  * It accepts a module loader and the astro settings, and it attempts to load the middlewares defined in the configuration.
  *
  * If not middlewares were not set, the function returns an empty array.
  */
-export async function loadMiddleware(moduleLoader: ModuleLoader) {
+export async function loadMiddleware(moduleLoader: ModuleLoader, logger: Logger) {
 	try {
 		const module = await moduleLoader.import(MIDDLEWARE_MODULE_ID);
 		return module;
-	} catch {
+	} catch (e: any) {
+		logger.error(
```

Comment:
Is there a specific reason for this to be a log instead of an actual throw?

Reply:
No particular reason. Should we throw an astro error? What do you think?

Reply:
I'd say so! Presumably if the middleware fails to load, we'd like the build to fail? I'd just wonder in dev if it would work correctly? Would it crash the dev server? If so, maybe it'd be possible to handle it correctly using the method Matthew added not too long ago.